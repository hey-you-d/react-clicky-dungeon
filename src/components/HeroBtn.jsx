import React from 'react';
import PropTypes from 'prop-types';

const HeroBtn = props => {
  const { onClickHandler } = props;
  const facialExpressions = {
    default: ["( '_' )", "('_'  )", "(  '_')"],
    hurt: '( >_< )',
    happy: ['( ^_^ )', '(^_^  )', '(  ^_^)'],
    unconscious: '(X _ X)'
  };

  return (
    <div
      role="button"
      tabIndex={-1}
      className="ingame-uber-box ingame-profilepic-box"
      onClick={onClickHandler}
      onKeyPress={onClickHandler}
    >
      <div>
        <div>{facialExpressions.default[0]}</div>
      </div>
    </div>
  );
};

HeroBtn.propTypes = {
  onClickHandler: PropTypes.func.isRequired
};

export default HeroBtn;
