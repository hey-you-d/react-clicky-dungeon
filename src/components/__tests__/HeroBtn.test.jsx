import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HeroBtn from '../HeroBtn';

configure({ adapter: new Adapter() });

describe('<HeroBtn />', () => {
  let wrapper;
  let mockClickHandler;

  beforeEach(() => {
    mockClickHandler = jest.fn();
  });

  it('should render properly', () => {
    wrapper = shallow(<HeroBtn onClickHandler={mockClickHandler} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the function specified in onClick attribute', () => {
    wrapper = shallow(<HeroBtn onClickHandler={mockClickHandler} />);

    wrapper.find('div.ingame-profilepic-box').simulate('click');

    expect(mockClickHandler).toHaveBeenCalled();
  });
});
