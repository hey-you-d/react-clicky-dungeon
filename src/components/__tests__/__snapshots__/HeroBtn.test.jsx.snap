// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBtn /> should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroBtn
    onClickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div>
          ( '_' )
        </div>
      </div>,
      "className": "ingame-uber-box ingame-profilepic-box",
      "onClick": [MockFunction],
      "onKeyPress": [MockFunction],
      "role": "button",
      "tabIndex": -1,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          ( '_' )
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "( '_' )",
        },
        "ref": null,
        "rendered": "( '_' )",
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div>
            ( '_' )
          </div>
        </div>,
        "className": "ingame-uber-box ingame-profilepic-box",
        "onClick": [MockFunction],
        "onKeyPress": [MockFunction],
        "role": "button",
        "tabIndex": -1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            ( '_' )
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "( '_' )",
          },
          "ref": null,
          "rendered": "( '_' )",
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
